uptane-standard.md

```diff
@@ -240,28 +277,222 @@ The Time Server SHALL expose a public interface allowing primaries to communicat
 
-#### Download and verify images
+The primary SHALL download metadata for all targets and perform a full verification on it as specified in {{full_verification}}.
 
-#### Send latest time to secondaries
+#### Download and verify images {#download_images_primary}
 
-#### Send metadata to secondaries
+The primary SHALL download and verify images for itself and for all of its associated secondaries. Images SHALL be verified by checking that the hash of the image file matches the hash specified in the director's targets metadata for that image.
 
-#### Send images to secondaries
+There may be several different filenames that all refer to the same image binary, as described in {{targets_meta}}. The primary SHALL associate each image binary with each of its possible filenames.
+
+#### Send latest time to secondaries {#send_time_primary}
+
+The primary SHALL send the time server's latest attested time to each ECU. The secondary SHALL verify the time message, then overwrite its current time with the received time.
+
+#### Send metadata to secondaries {#send_metadata_primary}
+
+The primary SHALL send the latest metadata it has downloaded to all of its associated secondaries.
+
+Full verification secondaries SHALL keep a complete copy of all metadata. A partial verification secondary SHALL keep *only* the targets metadata file from the director repository.
+
+#### Send images to secondaries {#send_images_primary}
+
+The primary SHALL send the latest image to each of its associated secondaries that have storage to receive it.
+
+For secondaries without storage, the primary SHOULD wait for a request from the secondary to stream the new image file to it. The secondary will send the request once it has verified the metadata sent in the previous step.
 
 ### Installing images on ECUs
 
-### Metadata verification
+Before installing a new image, an ECU SHALL perform the following five steps:
+
+1. Verify latest attested time ({{verify_time}})
+1. Verify metadata ({{verify_metadata}})
+1. Download latest image ({{download_image}})
+1. Verify image ({{verify_image}})
+1. Create and send version report ({{create_version_report}})
+
+
+#### Verify latest attested time {#verify_time}
+
+The ECU SHALL verify the latest downloaded time. To do so, it must:
+
+1. Verify that the signatures on the downloaded time are valid,
+2. Verify that the list of nonces/tokens in the downloaded time includes the token that the ECU sent in its previous version report
+3. Verify that the time downloaded is greater than the previous time
+
+If all three steps complete without error, the ECU SHALL overwrite its current attested time with the time it has just downloaded and generate a new nonce/token for the next request to the time server.
+
+If any check fails, the ECU SHALL NOT overwrite its current attested time, and SHALL jump to the fifth step ({{create_version_report}}). The ECU SHOULD reuse its previous token for the next request to the time server.
```

Comment:
"The ECU MUST reuse", right? What other option would there be? Using the new one seems like it would be confusing.